Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java

```diff
@@ -118,8 +118,9 @@ public boolean checkTrigger(GameEvent event, Game game) {
         }
         
         Permanent permanent = (Permanent) game.getLastKnownInformation(event.getTargetId(), Zone.BATTLEFIELD);
+        Player player = game.getPlayer(source.getControllerId());
         // A creature you control
-        if (!permanent.getControllerId().equals(this.getControllerId()) || !permanent.isCreature()) {
+        if (player == null || !player.equals(this.getControllerId()) || permanent == null || !permanent.isCreature()) {
```

Comment:
`player.equals(this.getControllerId())` -- that' correct, but not now (player class do not compare by uuid -- see #4407). Need `player.getId().equals(this.getControllerId())`

Reply:
Whoops, got it. Was too tired to make it happen last night.